packages/grid/x-data-grid/src/useDataGridProps.ts

```diff
@@ -42,14 +93,14 @@ export const useDataGridProps = (inProps: DataGridProps) => {
     const overrides = themedProps.components;
 
     if (!overrides) {
-      return { ...DEFAULT_GRID_SLOTS_COMPONENTS };
+      return { ...DATA_GRID_DEFAULT_SLOTS_COMPONENTS };
     }
 
     const mergedComponents = {} as GridSlotsComponent;
 
-    Object.keys(DEFAULT_GRID_SLOTS_COMPONENTS).forEach((key) => {
+    Object.keys(DATA_GRID_DEFAULT_SLOTS_COMPONENTS).forEach((key) => {
```

Comment:
Not about the PR, but I'm wondering why using a loop instead of 
```js
mergedComponents = {
  ...DATA_GRID_DEFAULT_SLOTS_COMPONENTS,
  ...overrides
}
```

Reply:
I think I did it to make `null` override works (if you want to disable the sort icon for instance).
But it works well without it, so it's probably a mistake on my side